api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -78,4 +79,69 @@ public PatientQueue getIncompletePatientQueue(Patient patient, Location location
 		return dao.getIncompletePatientQueue(patient, locationTo);
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getMostRecentQueue(org.openmrs.Patient)
+	 */
+	@Override
+	public PatientQueue getMostRecentQueue(Patient patient) {
+		return dao.getMostRecentQueue(patient);
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#assignVisitNumber(org.openmrs.module.patientqueueing.model.PatientQueue)
+	 */
+	@Override
+	public PatientQueue assignVisitNumber(PatientQueue patientQueue) {
+		Date today = new Date();
+		List<PatientQueue> patientQueueList = getPatientQueueList(null, OpenmrsUtil.firstSecondOfDay(today),
+		    OpenmrsUtil.getLastMomentOfDay(today), null, null, patientQueue.getPatient(), null);
+		
+		if (!patientQueueList.isEmpty() && patientQueueList.get(0).getVisitNumber() != null) {
+			patientQueue.setVisitNumber(patientQueueList.get(0).getVisitNumber());
+		} else {
+			patientQueue.setVisitNumber(generateVisitNumber(patientQueue.getLocationFrom(), patientQueue.getPatient()));
+		}
+		return patientQueue;
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#generateVisitNumber(org.openmrs.Location,
+	 *      org.openmrs.Patient)
+	 */
+	public String generateVisitNumber(Location location, Patient patient) {
+		
+		Date today = new Date();
+		
+		SimpleDateFormat formatterExt = new SimpleDateFormat(Context.getAdministrationService().getGlobalProperty(
+		    "patientqueueing.defaultDateFormat"));
+		
+		List<PatientQueue> patientQueues = getPatientQueueList(null, OpenmrsUtil.firstSecondOfDay(today),
+		    OpenmrsUtil.getLastMomentOfDay(today), null, location, null, null);
+		
+		int nextNumberInQueue = 1;
+		if (!patientQueues.isEmpty()) {
+			String currentVisitNumber = patientQueues.get(0).getVisitNumber();
+			nextNumberInQueue = Integer.parseInt(String.valueOf(currentVisitNumber.subSequence(currentVisitNumber.length(),
+			    -3)));
+		}
+		
+		String dateString = formatterExt.format(today);
+		
+		String locationName = location.getName();
+		
+		if (locationName.length() > 3) {
+			locationName = locationName.substring(0, 3);
+		}
+		
+		String zeroesToAppend;
+		if (nextNumberInQueue <= 9) {
+			zeroesToAppend = "00";
+		} else if (nextNumberInQueue < 100) {
+			zeroesToAppend = "0";
+		} else
```

Comment:
You can get rid of this else clause by assigning the empty string on variable declaration.